ilastik/workflows/carving/carvingGui.py

```diff
@@ -240,17 +240,14 @@ def findNextPrefixNumber(self):
 
         return last + 1
 
-    def saveAsDialog(self, name=""):
-        """special functionality: reject names given to other objects"""
-        namesInUse = self.getObjectNames()
-
+    def saveAsDialog(self, name_input_default: str, existing_names: List[str]):
         def generateObjectName():
```

Comment:
I think moving `generateObjectName` into `findNextPrefixNumber` would make sense here: we probably only care about generating fresh, unique names.

Also, if you move `name_input_default = name_input_default or generateObjectName()` out of the dialog, it would make the dialog independent of the enclosing class.